client/landing/jetpack-cloud/sections/backups/download/index.tsx

```diff
@@ -0,0 +1,107 @@
+/**
+ * External dependencies
+ */
+import React, { useCallback } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { BackupProgress } from './types';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { getSiteTitle } from 'state/sites/selectors';
+import { rewindBackup } from 'state/activity-log/actions';
+import { useLocalizedMoment } from 'components/localized-moment';
+import Confirm from './confirm';
+import getBackupProgressForRewindId from 'state/selectors/get-backup-progress-for-rewind-id';
+import getRewindBackupProgressRequestStatus from 'state/selectors/get-rewind-backup-progress-request-status';
+import InProgress from './in-progress';
+import QueryRewindBackupStatus from 'components/data/query-rewind-backup-status';
+import Ready from './ready';
+import Placeholder from './placeholder';
+import Error from './error';
+
+interface Props {
+	rewindId: string;
+}
+
+const BackupRestorePage = ( { rewindId }: Props ) => {
+	const dispatch = useDispatch();
+
+	const moment = useLocalizedMoment();
+	const backupDateString: string = moment.unix( rewindId ).format( 'LLL' );
+	const longBackupDateString: string = moment.unix( rewindId ).format( 'LLLL' );
+	const siteId = useSelector( getSelectedSiteId );
+	const siteTitle = useSelector( state => getSiteTitle( state, siteId ) );
+
+	const backupProgressRequestStatus: string | null = useSelector( state =>
+		getRewindBackupProgressRequestStatus( state, siteId )
+	);
+
+	const backupProgress: BackupProgress | null = useSelector( state =>
+		getBackupProgressForRewindId( state, siteId, rewindId )
+	);
+
+	const downloadUrl = backupProgress?.url;
+	const requestDownload = useCallback( () => {
+		if ( siteId && rewindId ) {
+			dispatch( rewindBackup( siteId, rewindId, {} ) );
+		}
+	}, [ dispatch, rewindId, siteId ] );
+
+	const onConfirm = () => {
+		requestDownload();
+	};
+
+	// The QueryRewindBackupStatus component will poll whenever there is a download id, so only retrieve it if the download is in progress
+	const downloadId =
+		backupProgress && ! isNaN( backupProgress.progress ) ? backupProgress.downloadId : null;
+
+	const render = () => {
+		// there is no backup download creation info
+		if ( null === backupProgress ) {
+			// and that is because it hasn't been loaded yet
+			if ( ! backupProgressRequestStatus || backupProgressRequestStatus !== 'success' ) {
```

Comment:
Would this result in the placehoder always showing up if the status is not `success`?

Reply:
the success is of the request for the progress of the backup, not the request of the backup. We want to hold rendering the page in this case ( and break the reactivity guidelines of calypso ) because depending on if there is a backup already completed the page could look very different.

also this should all be made much clearer via comments, that one is on me